drivers/i2c/meson/i2c.c

```diff
@@ -557,8 +593,8 @@ static void handle_response(void)
     bool write_error = i2c_get_error(regs, &bytes_read, &curr_token);
 
     // Prepare to extract data from the interface.
-    // Due to the way the token chain abstraction is done in the request buffer there will always be
-    // at least 1 more byte of request data then response data.
+    // INVARIANT: request data is always smaller than returned data to allow
+    //            reuse.
```

Comment:
wdym by this?

Reply:
The size of the buffer containing the request is expected to always be smaller than the size of data to be read

Reply:
wdym by to allow for reuse?